### PR TITLE
[DEV-7216] PageHeader Refactor for Greater Consumer Flexibility in Toolbar

### PR DESCRIPTION
**Pull request by MaxwellKendall**
- Allowing consumer to pass components rather than just component props
for greater flexibility in what is in the toolbar.

- Updating specificity for css selectors to maintain styles

### Author
- [x] Story Book has been rebuilt (new build artifacts in `/docs`)
- [x] Components have been rebuilt (new build artifacts in `/dist`)

👆 These can be done by executing `npm run build` and committing the build artifacts.

`N/A`  All new components have been exported in `/index.js` `(if applicable)`
`N/A` Design Review Complete
- [x] Relevant Draft release notes have been updated describing these changes ([use this template](https://github.com/fedspendingtransparency/data-transparency-ui/blob/master/release_notes_template.md) to document your changes) `(if applicable)`
`N/A` Integration Status Updated on relevant documentation page `(if applicable)`
- [x] `index.d.ts` updated with new prop types defined for new components. `(if applicable)`

### Reviewer
- [ ] Code review

### CODE REVIEW

File: components/PageHeader.jsx [hunk 1]
```diff
@@ -83,64 +33,30 @@ const PageHeader = ({
...
     return (
-        <main id={id} className={`usda-page__container${classNames ? ` ${classNames}` : ''}`}>
-            <section className={`usda-page-header${stickyClass}`} ref={stickyHeader}>
-                <div className="usda-page-header__container">
-                    <div className="usda-page-header__header">
-                        {overLine && <strong className="usda-page-header__overline">{overLine}</strong>}
-                        <div className="usda-page-header__title">
-                            <h1>{title}</h1>
-                        </div>
+        <section className={`usda-page-header${stickyClass}`} ref={stickyHeader}>
```
Comment (brett-varney) [inline components/PageHeader.jsx:37]: Why is this section tag instead of header?

File: components/PageHeader.jsx [hunk 2]
```diff
@@ -83,64 +33,30 @@ const PageHeader = ({
...
     const stickyClass = isSticky ? ' usda-page-header--sticky' : '';
 
-    const showToolbar = (
-        !isEmpty(fyProps) ||
-        !isEmpty(shareProps) ||
-        !isEmpty(downloadProps)
-    );
-
     return (
-        <main id={id} className={`usda-page__container${classNames ? ` ${classNames}` : ''}`}>
```
Comment (brett-varney) [inline components/PageHeader.jsx:93]: Thank you for removing main tag here!

**Review by brett-varney — COMMENTED** [review]
just a quick question

File: components/PageHeader.jsx (hunk 2 above)
Comment (MaxwellKendall) [reply components/PageHeader.jsx:93]: Yes this was a bug I introduced last sprint

File: components/PageHeader.jsx (hunk 1 above)
Comment (MaxwellKendall) [reply components/PageHeader.jsx:37]: We already have a header:

image

Comment (brett-varney) [reply components/PageHeader.jsx:37]: True, for the page template. This looks like a header for the "actual" page as well.

**Review by brett-varney — APPROVED** [review]

**Pull request closed without merging**